sdkproject/Assets/Mapbox/Unity/Utilities/HTTPRequest.cs

```diff
@@ -57,26 +55,13 @@ public void Cancel()
 
 		private IEnumerator DoRequest()
 		{
-			_request.Send();
-
-			DateTime timeout = DateTime.Now.AddSeconds(_timeout);
-			bool didTimeout = false;
-
-			while (!_request.isDone)
-			{
-				yield return null;
-				if (DateTime.Now > timeout)
-				{
-					_request.Abort();
-					didTimeout = true;
-					break;
-				}
-			}
+			yield return _request.SendWebRequest();
 
 			Response response;
-			if (didTimeout)
+			if (_request.isHttpError || _request.isNetworkError)
```

Comment:
Do we need an if check anymore?

Reply:
Yes, I think so. Or refactor the following lines a bit to account for cancelled requests.

Reply:
`response = Response.FromWebResponse(this, _request, _wasCancelled ? new Exception("Request Cancelled") : null);` 

This covers all cases, right?